### PR TITLE
fix: Replace useDispatch w. useQuery/request for Investigations module (src/Components/Facility/Investigations)

### LINKED ISSUE #7168: Replace `useDispatch` w. `useQuery`/`request`: Investigations (`src/Components/Facility/Investigations`)

### PR DESCRIPTION
**Pull request by aeswibon**
# Bug Fix

## Proposed Changes

- [x] Fixes #7168

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/Investigations/InvestigationSuggestions.tsx [hunk 1]
```diff
@@ -1,46 +1,54 @@
...
   const {
     consultationId,
     logUrl,
     investigations: previousInvestigations,
   } = props;
-  const dispatch = useDispatch<any>();
 
   const [investigations, setInvestigations] = useState<
     InvestigationType[] | null
   >(null);
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/InvestigationSuggestions.tsx]:
```suggestion
```

File: src/Components/Facility/Investigations/InvestigationSuggestions.tsx [hunk 2]
```diff
@@ -1,46 +1,54 @@
...
+  const { loading } = useQuery(routes.getConsultation, {
+    pathParams: {
+      id: consultationId,
+    },
+    onResponse: (res) => {
+      if (res && res.data) {
+        setInvestigations(res.data.investigation || []);
+      }
+    },
+  });
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/InvestigationSuggestions.tsx]: Remove the `setInvestigations` state as useQuery already manages a state for the response body.

instead do: 

```
const { data: investigations, loading } = useQuery(...);
```

File: src/Components/Facility/Investigations/index.tsx [hunk 3]
```diff
@@ -112,16 +107,43 @@ const Investigation = (props: {
...
+    routes.getPatient,
+    {
+      pathParams: { id: patientId },
+      onResponse: (res) => {
+        if (res.data) {
+          setSession(new Date().toString());
+        }
+      },
+    }
+  );
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/index.tsx]: same here...

File: src/Components/Facility/Investigations/investigationsTab.tsx [hunk 4]
```diff
@@ -23,63 +19,42 @@ export default function InvestigationTab(props: {
...
+  const { loading: investigationSessionLoading } = useQuery(
+    routes.getInvestigationSessions,
+    {
+      pathParams: {
+        consultation_external_id: consultationId,
+      },
+      onResponse: (res) => {
         if (res && res.data) {
-          setInvestigationSessions(res.data.reverse());
+          setInvestigationSessions(res.data);
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/investigationsTab.tsx]: remove this state here too

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@aeswibon `src/Components/Facility/Investigations/Reports/**` has not been migrated

image

**Comment by aeswibon** [comment]
> @aeswibon `src/Components/Facility/Investigations/Reports/**` has not been migrated
> 
> image

@rithviknishad I am working on it..

**Comment by aeswibon** [comment]
@rithviknishad Facing difficulties in migrating the `src/Components/Facility/Investigations/Reports/**`.

**Comment by rithviknishad** [comment]
What's the issue?

**Comment by aeswibon** [comment]
> What's the issue?

@rithviknishad Facing difficulty in converting handleLoadMore function. Can you help me in that. Just pushed the bug commit...

File: src/Redux/api.tsx [hunk 5]
```diff
@@ -1006,10 +1017,19 @@ const routes = {
...
   },
   editInvestigation: {
     path: "/api/v1/consultation/{consultation_external_id}/investigation/batchUpdate/",
     method: "PUT",
+    TRes: Type<Record<string, never>>(),
+    TBody: Type<{
+      investigations: {
+        external_id: any;
+        value: any;
+        notes: any;
```
Comment (rithviknishad) [inline src/Redux/api.tsx]: replace `any` with proper types. Avoid using any if we know for sure what these types are going to be

File: src/Redux/api.tsx [hunk 6]
```diff
@@ -993,10 +994,20 @@ const routes = {
...
     path: "/api/v1/consultation/{consultation_external_id}/investigation/",
     method: "POST",
+    TRes: Type<Record<string, never>>(),
+    TBody: Type<{
+      investigations: {
+        investigation: string;
+        value: any;
+        notes: any;
+        session: string;
+      }[];
```
Comment (rithviknishad) [inline src/Redux/api.tsx]: Same here, avoid `any`

**Review by rithviknishad — CHANGES_REQUESTED** [review]
lgtm, just a minor change

File: src/Redux/api.tsx (hunk 5 above)
Comment (aeswibon) [reply src/Redux/api.tsx]: Done

File: src/Redux/api.tsx (hunk 6 above)
Comment (aeswibon) [reply src/Redux/api.tsx]: Done

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
@aeswibon can you fix the merge conflict

**Comment by nihal467** [comment]
@aeswibon 
image

- [x] the investigation suggestion mentioned in the consultation form is not visible under the investigation section

cc: @rithviknishad @Ashesh3

**Comment by nihal467** [comment]
@aeswibon clear the merge conflict

**Comment by nihal467** [comment]
@aeswibon the cypress is fully failing, fix it

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**